enumeratum-cats/src/main/scala/enumeratum/values/CatsValueEnum.scala

```diff
@@ -0,0 +1,28 @@
+package enumeratum.values
+
+import cats.{Eq, Order, Show}
+
+trait CatsValueEnum[ValueType, EntryType <: ValueEnumEntry[ValueType]] {
+  this: ValueEnum[ValueType, EntryType] =>
+
+  implicit val eqInstance: Eq[EntryType] = Eq.fromUniversalEquals[EntryType]
```

Comment:
Would it make sense to have these depend on `Eq` on the `ValueType` and build an `Eq` for the `EntryType` using that?

Reply:
Interesting question - I tend to say no. The idea of an enum is having distinct values and these are guaranteed (correct me if I'm wrong) by the way the macros work, requiring literals in the definition. So the only valid `Eq` instance would be the one differentiating all enum values and we achieve that already by `fromUniversalEquals`. One can still have a custom instance by overriding this one, it's a one-liner.

So I'd leave it as-is for now, but arguments welcome :)